app/modules/academics/services.py

```
@@ -30,11 +39,18 @@ def get_level_by_id(self, school_id, level_id):
         if not level: raise HTTPException(status.HTTP_404_NOT_FOUND, "Level not found")
         return level
     def create_level(self, school_id, data):
-        if self.repository.get_level_by_code(school_id, data.code): raise HTTPException(status.HTTP_409_CONFLICT, "A level with this code already exists.")
+        code = data.code.strip()
+        if self.repository.get_level_by_code(school_id, code): raise HTTPException(status.HTTP_409_CONFLICT, "A level with this code already exists.")
```

Ruff (0.16.1)

[error] Multiple statements on one line (colon)

(E701)